csrc/mmdeploy/core/CMakeLists.txt

```diff
@@ -54,6 +54,11 @@ target_include_directories(${PROJECT_NAME}
         $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/third_party/json>
         )
 
+if (MSVC)
+    target_compile_options(${PROJECT_NAME} PUBLIC
+            $<$<COMPILE_LANGUAGE:CXX>:/Zc:preprocessor>)
+endif ()
```

Comment:
We'd better also move `add_compile_options($<$<COMPILE_LANGUAGE:CXX>:/Zc:__cplusplus>)` here since `__cplusplus` is may not be correctly defined on legacy MSVC without the flag.